core/src/main/java/org/jruby/javasupport/ext/JavaLang.java

```diff
@@ -340,6 +344,26 @@ public static IRubyObject integer_p(final ThreadContext context, final IRubyObje
                                                     val instanceof java.math.BigInteger);
         }
 
+        @JRubyMethod(name = "coerce")
+        public static IRubyObject coerce(final ThreadContext context, final IRubyObject self, final IRubyObject type) {
+            java.lang.Number val = (java.lang.Number) self.toJava(java.lang.Number.class);
+
+            // NOTE: a basic stub that always coverts Java numbers to Ruby ones (for simplicity)
+            // gist being this is not expected to be used heavily, if so should get special care
+            final IRubyObject value;
+            if (val instanceof java.math.BigDecimal) {
+                final RubyClass klass = context.runtime.getClass("BigDecimal");
+                if (klass == null) { // user should require 'bigdecimal'
```

Comment:
It seems like many more people would use BigDecimal than Complex to me...I'll ping MRI folks about it.